packages/kbn-discover-utils/src/components/custom_control_columns/degraded_docs_control.tsx

```diff
@@ -52,49 +56,52 @@ const degradedDocButtonLabelWhenNotPresent = i18n.translate(
   { defaultMessage: 'All fields in this document were parsed correctly' }
 );
 
-const degradedDocButtonLabelWhenDisabled = i18n.translate(
-  'discover.customControl.degradedDocDisabled',
-  {
-    defaultMessage:
-      'Degraded document field detection is currently disabled for this search. To enable it, include the METADATA directive for the `_ignored` field in your ES|QL query. For example:',
-  }
+const updateEsqlQuery = i18n.translate('discover.customControl.degradedDoc.updateEsqlQuery', {
+  defaultMessage: 'Update ES|QL query',
+});
+
+const formattedCTAMessage = (
+  <FormattedMessage
+    id="discover.customControl.degradedDocDisabled"
+    defaultMessage="Degraded document field detection is disabled for this search. To enable it, add {directive} to your ES|QL query."
+    values={{
+      directive: <EuiCode css={{ display: 'inline-block' }}>METADATA _ignored</EuiCode>,
+    }}
+  />
 );
 
 const DegradedDocs = ({
   Control,
   enabled = true,
+  addIgnoredMetadataToQuery,
   rowProps: { record },
   ...props
 }: {
   Control: RowControlComponent;
   rowProps: RowControlRowProps;
 } & DegradedDocsControlProps) => {
   const isDegradedDocumentExists = DEGRADED_DOCS_FIELDS.some(
-    (field) => field in record.raw && record.raw[field] !== null
+    (field) => field in record.raw && record.raw[field] !== null && record.raw[field] !== undefined
   );
 
   if (!enabled) {
-    const codeSample = 'FROM logs-* METADATA _ignored';
-
-    const tooltipContent = (
-      <div>
-        {degradedDocButtonLabelWhenDisabled}
-        <EuiSpacer size="s" />
-        <EuiCode>{codeSample}</EuiCode>
-      </div>
-    );
-
-    return (
-      <Control
-        disabled
-        data-test-subj="docTableDegradedDocDisabled"
-        tooltipContent={tooltipContent}
-        label={`${degradedDocButtonLabelWhenDisabled} ${codeSample}`}
-        iconType="indexClose"
-        onClick={undefined}
-        {...props}
-      />
-    );
+    if (addIgnoredMetadataToQuery) {
+      return (
+        <EnableESQLDegradedDocsControl addIgnoredMetadataToQuery={addIgnoredMetadataToQuery} />
+      );
+    } else {
+      return (
+        <Control
+          disabled
+          data-test-subj="docTableDegradedDocDisabled"
+          tooltipContent={formattedCTAMessage}
+          label={actionsHeaderAriaLabelDegradedAction}
+          iconType="indexClose"
+          onClick={undefined}
+          {...props}
+        />
+      );
+    }
```

Comment:
Would it be possible to have different icon colors when all good vs missing _ignored in ES|QL query? Currently, actions look exactly the same:

<img width="349" alt="Screenshot 2024-10-11 at 10 24 35" src="https://github.com/user-attachments/assets/9223e1bd-c353-4710-bf9e-c6525c0cf0d1">
<img width="409" alt="Screenshot 2024-10-11 at 10 24 44" src="https://github.com/user-attachments/assets/9976ebe1-7745-42b6-adeb-e85b09100fd2">

Maybe we could color it green when all good?

Reply:
FYI @l-suarez

Reply:
Or we could consider using different icons.

Reply:
Thanks!